src/main.rs

```diff
@@ -233,25 +233,159 @@ fn color_for_level(lvl: &JobLogLevel) -> Color {
     }
 }
 
+struct LocalArtifactStore {
+    lava: Arc<ThrottledLava>,
+    masker: Arc<Masker>,
+}
+
+impl LocalArtifactStore {
+    pub fn new(lava: Arc<ThrottledLava>, masker: Arc<Masker>) -> Self {
+        Self { lava, masker }
+    }
+
+    pub fn get_log(
+        &self,
+        id: i64,
+    ) -> impl Stream<Item = Result<Bytes, std::io::Error>> + Unpin + '_ {
+        self.masker.mask_stream(Box::pin(
+            self.lava
+                .log(id)
+                .map(|x| x.raw())
+                .flatten_stream()
+                .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e)),
+        ))
+    }
+
+    pub fn get_junit(
+        &self,
+        id: i64,
+    ) -> impl Stream<Item = Result<Bytes, std::io::Error>> + Unpin + '_ {
+        Box::pin(
+            self.lava
+                .job_results(id)
+                .into_stream()
+                .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e)),
+        )
+    }
+
+    pub fn get_full_junit(
+        &self,
+        ids: Vec<i64>,
+    ) -> impl Stream<Item = Result<Bytes, std::io::Error>> + Unpin + '_ {
+        Box::pin(self.get_junit(ids[0]))
```

Comment:
This seems a bit fishy? why havea full_junit concept when it only gets the junit for the first id?

Reply:
The issue here is that aggregating junit files in rust turns out to be messy, and it looked like a rabbit hole. So, in the spirit of incremental delivery ;) I just stubbed it to work in the vastly most common case (i.e. for `submit` type jobs). Then, later, as needed, we can expand it to collect all the junit results together into the aggregate you ideally want. This seems quite desirable, since I think the pipelines look better just referring to "junit.xml"; if we do this we can update the all pipelines mark that file as a report, and we'll get proper junit integration for monitor jobs automatically if/when we fix it up later.

Would you prefer I removed it?

Reply:
refreshing my mind on all this; It's a bit awkward as in from the user point of view if i got it right only jobs with a single lava job will current get a `junit.xml`  and if there are multiple jobs then that file is not availble.

That said according to the [gitlab documentation](https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportsjunit):
```
Some JUnit tools export to multiple XML files. You can specify multiple test report paths in a single job to concatenate them into a single file. Use either:

A filename pattern (junit: rspec-*.xml).
an array of filenames (junit: [rspec-1.xml, rspec-2.xml, rspec-3.xml]).
A Combination of both (junit: [rspec.xml, test-results/TEST-*.xml]).
```

So it seems to me we could get rid of the Full junit and let gitlab deal with the concatenation ? Or did you see issues with that?